### PR TITLE
Task/253: .NET 6 Migration

### LINKED ISSUE #253: Update .NET
Currently .NET core 3.1 is used, it would be prudent to update to the newest version before support for 3.1 is dropped.

### PR DESCRIPTION
**Pull request by malgreen**
# Description

Fixes #253
- Target dotnet framework in all 3 .csproj files updated to *net6.0* from *netcoreapp3.1*
- Updated some NuGet packages to align with new .NET version
- Updated startup configuration for the server because of breaking changes to the new packages
- Fixed unit tests, all should now pass
- Updated docker files
- Updated GitHub actions to .NET 6
- Updated README

*NOTICE*: This change requires developers to update their .NET SDK & runtime to .NET 6.0, this means that the install guides also need to be updated, which is covered in another PR: https://github.com/aau-giraf/.github/pull/8

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] Unit tests on multiple machines with .NET SDK & runtime updated

**Development Configuration**
* SDK: .NET 6.0

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation, if necessary
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device.

### CODE REVIEW

File: GirafRest.Test/FakeRepositoryContexts/FakeWeekRepositoryContext.cs [hunk 1]
```diff
@@ -37,6 +37,7 @@ private void seed()
                 //girafuser1 used for getAllWeeksOfUser
                 var girafuser1 = new GirafUser();
                 girafuser1.Id = "1";
+                girafuser1.DisplayName = "";
```
Comment (Lildhansen) [inline GirafRest.Test/FakeRepositoryContexts/FakeWeekRepositoryContext.cs:40]: Hvad har dette med .NET 6 migration at gøre?

File: GirafRest/Setup/Startup.cs [hunk 2]
```diff
@@ -87,6 +87,7 @@ public Startup(IWebHostEnvironment env)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddApplicationInsightsTelemetry();
+            services.AddSingleton<IProcessingStrategy, AsyncKeyLockProcessingStrategy>();
```
Comment (Lildhansen) [inline GirafRest/Setup/Startup.cs:90]: Hvad er funktionaliteten og hvorfor er det her nødvendigt?

File: GirafRest.Test/FakeRepositoryContexts/FakeWeekRepositoryContext.cs (hunk 1 above)
Comment (malgreen) [reply GirafRest.Test/FakeRepositoryContexts/FakeWeekRepositoryContext.cs:40]: Det var bare en fejl fra tidligere unit tests, som var sat forkert op - vi tænkte at vi ville fikse dem inden push.
Har ikke rigtigt noget med .NET6 at gøre.

File: GirafRest/Setup/Startup.cs (hunk 2 above)
Comment (malgreen) [reply GirafRest/Setup/Startup.cs:90]: Det fordi den nye version af AspNetCoreRateLimit ikke har en default strategy, så det skal manuelt defineres i startup af server.
[Se eventuelt her](https://stackoverflow.com/a/69157526)

**Review by Djepse20 — APPROVED** [review]
Looks good to me, cant see any problems.
Rate limiting seems to have no effect at the momenet, so might as well keep it.

**Review by Lildhansen — APPROVED** [review]
Det ser finno ud

**Pull request merged**